apps/src/StudioApp.js

```diff
@@ -1634,27 +1739,29 @@ StudioApp.prototype.handleHideSource_ = function (options) {
           appType: dashboard.project.getStandaloneApp()
         }), div);
       }
-    } else if (!options.embed && !dom.isMobile()) {
-      var runButton = document.getElementById('runButton');
-      var buttonRow = runButton.parentElement;
-      var openWorkspace = document.createElement('button');
-      openWorkspace.setAttribute('id', 'open-workspace');
-      openWorkspace.appendChild(document.createTextNode(msg.openWorkspace()));
-
-      dom.addClickTouchEvent(openWorkspace, function() {
-        // TODO: don't make assumptions about hideSource during init so this works.
-        // workspaceDiv.style.display = '';
-
-        // /c/ URLs go to /edit when we click open workspace.
-        // /project/ URLs we want to go to /view (which doesnt require login)
-        if (/^\/c\//.test(location.pathname)) {
-          location.href += '/edit';
-        } else {
-          location.href += '/view';
-        }
-      });
 
-      buttonRow.appendChild(openWorkspace);
+      if (!options.embed) {
+        var runButton = document.getElementById('runButton');
+        var buttonRow = runButton.parentElement;
+        var openWorkspace = document.createElement('button');
+        openWorkspace.setAttribute('id', 'open-workspace');
+        openWorkspace.appendChild(document.createTextNode(msg.openWorkspace()));
+
+        dom.addClickTouchEvent(openWorkspace, function () {
+          // TODO: don't make assumptions about hideSource during init so this works.
```

Comment:
Good question. It's been there a while. See https://github.com/code-dot-org/code-dot-org/commit/34a63e75f5335f4dd1bd663c5fceeea8f94f897c. @joshlory ?